### PR TITLE
resize string so it doesn't over flow

### PR DESCRIPTION
**Pull request by duke8253**
This fixes a crash in `traffic_dump`.

```
[ 00 ] traffic_server                       sdk_sanity_check_mbuffer                                                                                                                                       ( InkAPI.cc:605 )
[ 01 ] traffic_server                       TSMimeHdrFieldNext                                                                                                                                             ( InkAPI.cc:3417 )
[ 02 ] traffic_dump.so                      traffic_dump::TransactionData::write_message_node_no_content[abi:cxx11](tsapi_mbuffer*&, tsapi_mloc*&, std::basic_string_view<char, std::char_traits<char> >)  ( transaction_data.cc:270 )
[ 03 ] traffic_dump.so                      traffic_dump::TransactionData::write_client_request_node_no_content(tsapi_mbuffer*&, tsapi_mloc*&)                                                             ( transaction_data.cc:361 )
[ 04 ] traffic_dump.so                      traffic_dump::TransactionData::global_transaction_handler(tsapi_cont*, TSEvent, void*)                                                                         ( transaction_data.cc:466 )
[ 05 ] traffic_server                       INKContInternal::handle_event(int, void*)                                                                                                                      ( InkAPI.cc:1155 )
[ 06 ] traffic_server                       handleEvent                                                                                                                                                    ( I_Continuation.h:227 )
[ 07 ] traffic_server                       handleEvent                                                                                                                                                    ( I_Continuation.h:223 )
[ 08 ] traffic_server                       APIHook::invoke(int, void*) const                                                                                                                              ( InkAPI.cc:1392 )
[ 09 ] traffic_server                       HttpSM::state_api_callout(int, void*)                                                                                                                          ( HttpSM.cc:1573 )
[ 10 ] traffic_server                       HttpSM::state_read_client_request_header(int, void*)                                                                                                           ( HttpSM.cc:941 )
[ 11 ] traffic_server                       HttpSM::main_handler(int, void*)                                                                                                                               ( HttpSM.cc:2820 )
[ 12 ] traffic_server                       do_api_callout                                                                                                                                                 ( HttpSM.cc:423 )
[ 13 ] traffic_server                       do_api_callout                                                                                                                                                 ( HttpSM.cc:418 )
[ 14 ] traffic_server                       HttpSM::state_api_callout(int, void*)                                                                                                                          ( HttpSM.cc:1653 )
[ 15 ] traffic_server                       HttpSM::state_api_callback(int, void*)                                                                                                                         ( HttpSM.cc:1453 )
[ 16 ] traffic_server                       TSHttpTxnReenable                                                                                                                                              ( InkAPI.cc:6449 )
[ 17 ] txn_box.so                           Context::ts_callback(tsapi_cont*, TSEvent, void*)                                                                                                              ( Context.cc:377 )
[ 18 ] traffic_server                       INKContInternal::handle_event(int, void*)                                                                                                                      ( InkAPI.cc:1155 )
[ 19 ] traffic_server                       handleEvent                                                                                                                                                    ( I_Continuation.h:227 )
[ 20 ] traffic_server                       handleEvent                                                                                                                                                    ( I_Continuation.h:223 )
[ 21 ] traffic_server                       APIHook::invoke(int, void*) const                                                                                                                              ( InkAPI.cc:1392 )
[ 22 ] traffic_server                       HttpSM::state_api_callout(int, void*)                                                                                                                          ( HttpSM.cc:1573 )
[ 23 ] traffic_server                       HttpSM::state_api_callback(int, void*)                                                                                                                         ( HttpSM.cc:1453 )
[ 24 ] traffic_server                       TSHttpTxnReenable                                                                                                                                              ( InkAPI.cc:6449 )
[ 25 ] traffic_dump.so                      traffic_dump::TransactionData::global_transaction_handler(tsapi_cont*, TSEvent, void*)                                                                         ( transaction_data.cc:534 )
[ 26 ] traffic_server                       INKContInternal::handle_event(int, void*)                                                                                                                      ( InkAPI.cc:1155 )
[ 27 ] traffic_server                       handleEvent                                                                                                                                                    ( I_Continuation.h:227 )
[ 28 ] traffic_server                       handleEvent                                                                                                                                                    ( I_Continuation.h:223 )
[ 29 ] traffic_server                       APIHook::invoke(int, void*) const                                                                                                                              ( InkAPI.cc:1392 )
[ 30 ] traffic_server                       HttpSM::state_api_callout(int, void*)                                                                                                                          ( HttpSM.cc:1573 )
[ 31 ] traffic_server                       HttpSM::state_api_callback(int, void*)                                                                                                                         ( HttpSM.cc:1453 )
[ 32 ] traffic_server                       TSHttpTxnReenable                                                                                                                                              ( InkAPI.cc:6449 )
[ 33 ] athenz_principal_check.so (deleted)
```

### CODE REVIEW

**Review by bneradt — COMMENTED** [review]
Looks good to me. Thanks for the fix. I'll let a non-Yahoo committer approve.

**Comment by bneradt** [comment]
This is a very safe fix which addresses a crash. Marking for 9.2.x since it applies there as well.

**Review by randall — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.2.x

**Pull request merged**
